ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -146,10 +138,185 @@ const primalOutputStrings = {
+  if (typeof data.decOffset === 'undefined')
+    data.decOffset = parseInt(matches.id, 16) - firstHeadmarker;
+  // The leading zeroes are stripped when converting back to string, so we re-add them here.
+  // Fortunately, we don't have to worry about whether or not this is robust,
+  // since we know all the IDs that will be present in the encounter.
+  return '00' + (parseInt(matches.id, 16) - data.decOffset).toString(16).toUpperCase();
+};
+
+// For giant lasers.
+const numberOutputStrings = [0, 1, 2, 3, 4].map((n) => {
+  const str = n.toString();
+  return {
+    en: str,
+    de: str,
+    fr: str,
+    ja: str,
+    cn: str,
+    ko: str,
+  };
+});
+
 export default {
   zoneId: ZoneId.EdensPromiseEternitySavage,
   timelineFile: 'e12s.txt',
   triggers: [
+    {
+      // Headmarkers are randomized, so handle them all with a single trigger.
+      id: 'E12S Headmarker',
+      netRegex: NetRegexes.headMarker({}),
+      response: (data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          formlessBusterAndSwap: {
+            en: 'Tank Buster + Swap',
+          },
+          formlessBusterOnYOU: {
+            en: 'Tank Buster on YOU',
+          },
+          // The first round has only one blue.
+          titanBlueSingular: {
+            en: 'Blue Weight',
+          },
+          // The second and two rounds of bombs have a partner.
+          // The third is technically fixed by role with a standard party (one dps, one !dps),
+          // but call out your partner anyway in case you've got 8 blus or something.
+          titanBlueWithPartner: {
+            en: 'Blue (with ${player})',
+          },
+          titanOrangeStack: {
+            en: 'Orange Stack',
+          },
+          titanYellowSpread: {
+            en: 'Yellow Spread',
+          },
+          // This is sort of redundant, but if folks want to put "square" or something in the text,
+          // having these be separate would allow them to configure them separately.
+          square1: numberOutputStrings[1],
+          square2: numberOutputStrings[2],
+          square3: numberOutputStrings[3],
+          square4: numberOutputStrings[4],
+          triangle1: numberOutputStrings[1],
+          triangle2: numberOutputStrings[2],
+          triangle3: numberOutputStrings[3],
+          triangle4: numberOutputStrings[4],
+        };
+
+        const id = getHeadmarkerId(data, matches);
+
+        // Track tankbuster targets, regardless if this is on you or not.
+        // Use this to make more intelligent calls when the cast starts.
+        if (id === '00DA') {
+          data.formlessTargets = data.formlessTargets || [];
+          data.formlessTargets.push(matches.target);
+        } else if (id === '00BB') {
+          data.weightTargets = data.weightTargets || [];
+          data.weightTargets.push(matches.target);
+
+          // Handle double blue titan on 2nd and 3rd iterations.
```

Comment:
Third iteration I don't think your partner will ever matter, since it's a T/H and a DPS, so I imagine these spots aren't being called on the fly?

Reply:
```
          // The second and two rounds of bombs have a partner.
          // The third is technically fixed by role with a standard party (one dps, one !dps),
          // but call out your partner anyway in case you've got 8 blus or something.
```

It's kind of a "might as well call it out" scenario, since we're already doing the work.